src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -6074,24 +6074,43 @@ protected override bool IsItemContainer(string path)
         /// <param name="force">If true, force move the directory, overwriting anything at the destination.</param>
         private void MoveDirectoryInfoUnchecked(DirectoryInfo directory, string destinationPath, bool force)
         {
-            try
+            if (!IsSameWindowsVolume(directory.FullName, destinationPath))
             {
-                if (InternalTestHooks.ThrowExdevErrorOnMoveDirectory)
-                {
-                    throw new IOException("Invalid cross-device link", hresult: MOVE_FAILED_ERROR);
-                }
-
-                directory.MoveTo(destinationPath);
+                CopyAndDelete(directory, destinationPath, force);
```

Comment:
I again very wonder why we do the check if .Net DirectoryInfo.MoveTo() method already does this for us and throw IOException? https://github.com/dotnet/runtime/blob/82ca681cbac89d813a3ce397e0c665e6c051ed67/src/libraries/System.IO.FileSystem/src/System/IO/DirectoryInfo.cs#L212-L216
So why beat around the bush?
This method behaves the same for both Unix and Windows - no need to have two code paths.
We only need to exclude the fallback to CopyAndDelete() for FileNotFoundException, DirectoryNotFoundException, PathTooLongException, and SecurityException (not UnauthorizedAccessException!).
It may be a more aggressive fallback for unknown IO errors that wraps a driver errors - what's wrong with that if we use the same template anyway - try and fallback?

Reply:
I missed that, we can certainly check for that specific error instead of having this method

Reply:
What do you see dangerous in the call to CopyAndDelete() in the case of these IOException-s? In the best case, the tree will be moved, in the worst case, we will receive an exception and can return an error to the user.

Reply:
ok, looking at the .NET code, it seems reasonable to try CopyAndDelete for the 2 cases of IOException

Reply:
I think it makes no sense to fallback to CopyAndDelete for FileNotFoundException, DirectoryNotFoundException, PathTooLongException, SecurityException and UnauthorizedAccessException - it is better to re-throw and write an error message in MoveDirectoryInfoItem() as previously.